# AICB1. 챗GPT API 사용해 보기

In [1]:
!pip install openai

Defaulting to user installation because normal site-packages is not writeable


In [1]:
import os
from openai import OpenAI
from getpass import getpass

# os.environ['OPENAI_API_KEY'] = getpass()

In [9]:
try:
    from google.colab import userdata
    IN_COLAB = True
    OPENAI_KEY = userdata.get('API_KEY_OPENAI')
# except ImportError:
#     IN_COLAB = False
#     OPENAI_KEY = os.environ['API_KEY_OPENAI']
except ImportError:
    IN_COLAB = False
    OPENAI_KEY = os.environ['API_KEY_OPENAI']

In [19]:
# from google.colab import userdata

# userdata.get('OPENAI_API_KEY')

In [11]:
# from openai import OpenAI
# from google.colab import userdata
# client = OpenAI(api_key=userdata.get('OPENAI_API_KEY'))

In [13]:
client = OpenAI(api_key=OPENAI_KEY)

In [15]:
completion = client.chat.completions.create(
    model="gpt-3.5-turbo",
    messages=[
        {"role": "system", "content": "당신은쇼핑몰의 상담원입니다."},
        {"role": "user", "content": "안녕하세요!"}
    ]
)
completion.choices[0].message

RateLimitError: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}

### # RateLimitError: Error code: 429

- 이 오류는 OpenAI API를 사용할 때 요청 한도를 초과했을 때 발생하는 RateLimitError예요.
- 쉽게 말하면, “요금제에서 허용된 만큼 다 썼어요!”라는 뜻이죠.


In [ ]:
completion = client.chat.completions.create(
    model="gpt-3.5-turbo-1106",
    messages=[
        {"role": "system", "content": "당신은 헤이마트의 상담원입니다."},
        {"role": "user", "content": "저녁으로 무엇을 먹을까요? 짧게 설명해주세요."}
    ],
    temperature=1.8,
    max_tokens=100,
    top_p=1,
    presence_penalty=0,
    frequency_penalty=0,
    n=2,
    stop=None
)

for choice in completion.choices:
    print(choice.index, choice.message.content)

0 저녁으로 채식 요리인 시원한 그린 샐러드나 생선 스테이크와 채소를 곁들인 요푸라이요. 상쾌하고 건강한 선택입니다!
1 생선 요리를 먹을 것을 추천합니다. 생선은 건강에 좋은 비타민과 단백질이 풍부하며 가볍게 먹으면서 영양을 충분히 섭취할 수 있어 좋습니다. 또한 다양한 요리 방법으로 다채로
